fs_src/script.js

```diff
@@ -1624,6 +1658,30 @@ function setPreviewColor(c) {
   el(c, "hue_value").innerHTML = `${el(c, "hue").value}&#176;`;
   el(c, "saturation_value").innerHTML = `${el(c, "saturation").value}%`;
   el(c, "brightness_value").innerHTML = `${el(c, "brightness").value}%`;
+  el(c, "colortemperature_value").innerHTML = `${el(c, "colortemperature").value}mired`;
+}
+
+function clamprgb(val) {
+    let min = 0;
+    let max = 255;
+    return Math.max(min, Math.min(val, max))
+}
+
+function colortemp2rgb(t, v) {
+  //Formula by Tanner Helland
+  var temperature = 1000000.0 / t / 100.0;
+  var scale = 1 / 2.55;
+  
+  return [
+    (temperature <= 66 ? 255 :
```

Comment:
this is completely unreadable, please pull the individual components out of the array definition into more readable if/else statements. make sure you run `make format` too - we not clang-format js as well.

Reply:
It's not from this PR, this PR is based on [#cct](https://github.com/mongoose-os-apps/shelly-homekit/pull/853) please to not review before that one is merged and I requested a review.

Reply:
yeah, sorry abotu that, started with the wrong one.